batch/batch/front_end/front_end.py

```diff
@@ -684,11 +684,18 @@ async def insert(tx):
                         log.info(f'bunch containing job {(batch_id, jobs_args[0][1])} already inserted ({err})')
                         return
                     raise
-                await tx.execute_many('''
+                try:
+                    await tx.execute_many('''
 INSERT INTO `job_parents` (batch_id, job_id, parent_id)
 VALUES (%s, %s, %s);
 ''',
-                                      job_parents_args)
+                                          job_parents_args)
+                except pymysql.err.IntegrityError as err:
+                    # 1062 ER_DUP_ENTRY https://dev.mysql.com/doc/refman/5.7/en/server-error-reference.html#error_er_dup_entry
+                    if err.args[0] == 1062:
+                        raise web.HTTPBadRequest(
+                            text=f'bunch contains job with duplicated parents ({err})')
+                    raise
```

Comment:
This is just being caught by the exception handler below and batch 500s.

Reply:
good point...

Reply:
I added a catch for HTTPException which is re-raised without a wrapper exception